app/src/main/java/com/aerogear/androidshowcase/providers/PushServiceProvider.java

```diff
@@ -0,0 +1,57 @@
+package com.aerogear.androidshowcase.providers;
+
+import com.aerogear.androidshowcase.R;
+
+import org.aerogear.mobile.core.MobileCore;
+import org.aerogear.mobile.core.executor.AppExecutors;
+import org.aerogear.mobile.core.reactive.Responder;
+import org.aerogear.mobile.push.PushService;
+import org.aerogear.mobile.push.UnifiedPushConfig;
+
+import java.util.Arrays;
+
+public class PushServiceProvider {
+
+    private static PushServiceProvider instance;
+
+    private PushService pushService;
+    private Exception registrationException;
+
+    protected PushServiceProvider() {
+        pushService = new PushService.Builder().openshift().build();
+    }
+
+    /**
+     * Register the push service.
+     */
+    public void registerDevice() {
+        pushService.registerDevice().respondOn(new AppExecutors().mainThread())
+            .respondWith(new Responder<Boolean>() {
+                @Override
+                public void onResult(Boolean value) {
+                    MobileCore.getLogger().info("");
+                }
+
+                @Override
+                public void onException(Exception exception) {
+                    MobileCore.getLogger().error(exception.getMessage(), exception);
+                    registrationException = exception;
+                }
+            });
+    }
+
+    /**
+     * Retrieve the exception that occurred during registration, if any.
+     * @return The exception that occurred during registration, or null.
+     */
+    public Exception getRegistrationException() {
```

Comment:
So this is a weird situation.  If the push is taking a long time to resolve it is possible that the user might go to the push page before the registration fails and not see the message until they refresh the page.

The more correct solution would be to wrap this in some kind of reactive element, but that might be overkill for what is basically a small race condition.